File: crates/auto-lang/test/a2r/006_struct/struct.expected.rs
struct Point {
    x: i32,
    y: i32,
}

struct Circle {
    radius: f64,
    border: u32,
    center: Point,
}

fn main() {
    let mut p: Point = Point { x: 1, y: 2 };
    p.x = 3;
    print!("P: ", p.x, ", ", p.y);

    let circle: Circle = Circle { radius: 5.0, border: 1, center: Point { x: 50, y: 50 } };
    print!("C: ", circle.center.x, ", ", circle.center.y, ", ", circle.radius);
}

File: crates/auto-lang/test/a2c/02_types/001_struct/struct.expected.c
#include "struct.h"



int main(void) {
    struct Point p = {.x = 1, .y = 2};
    p.x = 3;
    printf("%s %d %s %d\n", "P: ", p.x, ", ", p.y);

    struct Circle circle = {.radius = 5.0, .border = 1, .center = {.x = 50, .y = 50}};
    printf("%s %d %s %d %s %d\n", "C: ", circle.center.x, ", ", circle.center.y, ", ", circle.radius);
    return 0;
}
